### PR TITLE
updated lca and example script

### PR DESCRIPTION
**Pull request by alishbaimran**
- Fixed linear classifier to include SMOTE for training and ability to save predictions for figure use.
- Complete example script that can be used for training classifier and generating % infected cells figure.
- Fixed merge error.

### CODE REVIEW

File: viscy/representation/lca.py [hunk 1]
```diff
@@ -168,6 +200,10 @@ def train_and_test_linear_classifier(
     split_ratio: tuple[int, int, int] = (0.4, 0.2, 0.4),
     batch_size: int = 1024,
     lr: float = 1e-3,
+    use_smote: bool = False,
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply viscy/representation/lca.py:203]: So effectively no SMOTE will be applied.

**Comment by ziw-liu** [comment]
@Soorya19Pradeep plans to remove the invalid 'background' class from annotations. In that case class balancing should no longer be necessary since the remaining binary classification is already (roughly) balanced. We also need to test if the infection ratio estimation will be improved without significantly overestimating 'background'.

**Comment by alishbaimran** [comment]
> @Soorya19Pradeep plans to remove the invalid 'background' class from annotations. In that case class balancing should no longer be necessary since the remaining binary classification is already (roughly) balanced. We also need to test if the infection ratio estimation will be improved without significantly overestimating 'background'.

@Soorya19Pradeep @ziw-liu @mattersoflight 

From my understanding, Soorya is just planning to remove these from the dataframe we load of the annotations csv to then plot the infection status on the UMAP/PCA. The actual annotations won't be changing and will still include background (only removed in the visualization). Please correct me if I'm wrong. I'm planning to train a logistic regression model with 80/20 split and use SMOTE (assuming background class is staying we will still need this) for computing metrics.

**Comment by ziw-liu** [comment]
@alishbaimran we decided to not use the samples labelled as background for training linear classifiers. This is in part because there is <1k examples in the test set and oversampling them more than 10 times will be noisy. The problem can then be viewed as two-step classification, where a segmentation algorithm (or an annotator) classifies cell vs background, and a binary classifier classifies only the cells based on a feature extractor (contrastive encoder).

**Comment by ziw-liu** [comment]
Replaced by #169.

**Pull request closed without merging**